soh/src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -559,8 +574,15 @@ void func_80AE3834(EnRd* this, PlayState* play) {
     Color_RGBA8 sp2C = D_80AE493C;
     Player* player = GET_PLAYER(play);
     s16 temp_v0 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y - this->unk_30E - this->unk_310;
+    
+    if (CVarGetInteger("gNoRedeadFreeze", 0)) {
+        if (!(this->unk_312 & 0x80)) {
+            player->actor.freezeTimer = 0;
+        }
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_AIM);
+        func_80AE2B90(this, play);
 
-    if (ABS(temp_v0) < 0x2008) {
+    } else if (ABS(temp_v0) < 0x2008) {
         if (!(this->unk_312 & 0x80)) {
```

Comment:
This line (in two spots) seems to be the actual branch condition that controls the freezing.
I wonder if you can remove the duplicated code blocks in favor of just changing these lines to something like this:
```suggestion
        if (!(this->unk_312 & 0x80) && !CVarGetInteger("gNoRedeadFreeze", 0)) {
```

Just make sure to verify that redeads in market vs everywhere else work the same when the cheat is off

Reply:
I will certainly give that a try. This is how I wanted to do it originally, but wasn't sure where the market code was, and then figured out the way I did find.

Reply:
Seems to work like a charm! Thanks Archez!